client/lorawan.go

```diff
@@ -4,64 +4,68 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
-	"errors"
+	"fmt"
 	"net/http"
+	"net/url"
+	"time"
 
 	"github.com/NickGowdy/deveui-cli/device"
 )
 
-// Client used to communicate to external services
-type Client interface {
-	Do(*http.Request) (resp *http.Response, err error)
-}
-
 // LoraWAN used to communicate to LoRaWAN external system
 type LoraWAN struct {
-	baseURL string
-	client  Client
-}
-
-func NewLoraWAN(baseURL string, client Client) *LoraWAN {
-	return &LoraWAN{
-		baseURL: baseURL,
-		client:  client,
-	}
+	fullURL *url.URL
+	client  *http.Client
 }
 
 const endpoint = "/sensor-onboarding-sample" // endpoint for saving DevEUI via LoRaWAN
 
-// RegisterDevice registers new device using LoraWAN external service
-func (l *LoraWAN) RegisterDevice(ctx context.Context) (*device.Device, error) {
-	device := device.NewDevice()
-	identifier := device.GetIdentifier()
-	b := new(bytes.Buffer)
-	reqBody := map[string]string{"Deveui": identifier}
-
-	err := json.NewEncoder(b).Encode(&reqBody)
+func NewLoraWAN(baseURL string, timeout time.Duration) (*LoraWAN, error) {
+	path, err := url.JoinPath(baseURL, endpoint)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to join url: %w", err)
```

Comment:
Wrapping errors can help to identify where an error came from, especially when multiple functions can return the same error message. The https://github.com/pkg/errors package can give you stack trace along with the error, and their .Wrap method is convenient to use, but I try not to use this package for new projects anymore as per the description in the repo roadmap.